### PR TITLE
Additional tests for processRadioSchedules

### LINKED ISSUE #5871: Add additional RadioSchedule tests
**Is your feature request related to a problem? Please describe.**
During code review we identified a few places in the RadioSchedule implementation in which it would be good to have some additional tests.

**Describe the solution you'd like**
- Add unit tests for `processRadioSchedule.js` https://github.com/bbc/simorgh/blob/739cf150394024da8fd983d1913959835d8d6bfa/src/app/containers/RadioSchedule/utilities/processRadioSchedule.js
- Add assertion unit tests to `RadioSchedule/Canonical/index.test.js` to test `RadioFrequencyLink` is displayed when it should be. https://github.com/bbc/simorgh/blob/739cf150394024da8fd983d1913959835d8d6bfa/src/app/containers/RadioSchedule/Canonical/index.test.jsx
  **This is also tested by a snapshot test**

**Testing notes**
- [ ] This feature is expected to need manual testing.

### PR DESCRIPTION
**Pull request by itimirichard**
Resolves https://github.com/bbc/simorgh/issues/5871

**Overall change:** 
_Adds assertions for the processRadionSchedule, getLink and getProgramState functions._

**Code changes:**
- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const currentTime = Date.now();
+    const startTime = Date.now() - 1000;
+    const endTime = Date.now() + 1000;
+
+    expect(getProgramState(currentTime, startTime, endTime)).toBe('live');
+  });
+  it('should return `onDemand` when currentTime is greater than endTime', () => {
+    const currentTime = Date.now();
+    const startTime = Date.now() - 500;
+    const endTime = Date.now() - 1000;
```
Comment (OlgaLyubin) [inline src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js]: It seems like the `endTime` here is smaller than `startTime`

Comment (itimirichard) [reply src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js]: Thanks @OlgaLyubin  should be fixed now.

File: src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      '/persian/bbc_dari_radio/p07zbtbf',
+    );
+  });
+});
+
+describe('getProgramState', () => {
+  it('should return `live` when currentTime is greater than startTime but less than endTime', () => {
+    const currentTime = Date.now();
+    const startTime = Date.now() - 1000;
+    const endTime = Date.now() + 1000;
```
Comment (sareh) [inline src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js]: Thanks for all of these unit tests.

I'd recommend updating all these tests to use the currentTime and increment that. The reason is, that from the time when the first Date.now() is run to the next time Date.now() is run, time elapses, so they won't have the same identical reference point.
```suggestion
    const startTime = currentTime - 1000;
    const endTime = currentTime + 1000;
```

Comment (itimirichard) [reply src/app/containers/RadioSchedule/utilities/processRadioSchedule.test.js]: Thanks @sareh . Those tests have now been updated

**Review by Bopchy — APPROVED** [review]

**Review by OlgaLyubin — APPROVED** [review]
Thank you for adding these, looks good 👍

**Comment by PriyaKR** [comment]
This PR doesn't require any testing from a tester.

**Review by sareh — APPROVED** [review]

**Pull request merged**